syntax = "proto3";

package common_services;

import "google/protobuf/timestamp.proto";

// Message structure for a log entry
message LogEntry {
  google.protobuf.Timestamp timestamp = 1;
  string level = 2; // e.g., INFO, DEBUG, ERROR
  string service_name = 3; // Name of the service generating the log
  string logger_name = 4; // Name of the logger (e.g., module name)
  string message = 5;
  map<string, string> metadata = 6; // For structured data like request IDs, etc.
}

// Request for streaming logs (can be extended with filters in the future)
message StreamLogsRequest {
  // string min_level = 1; // Example filter: minimum log level
  // string service_filter = 2; // Example filter: specific service if this was a central aggregator
}

// Service for streaming logs from a service instance
service LoggingStreamer {
  // Streams log entries as they are generated by the service.
  rpc StreamLogs (StreamLogsRequest) returns (stream LogEntry);
}
